manualdirectdebit.php

```diff
@@ -166,3 +166,103 @@ function manualdirectdebit_civicrm_navigationMenu(&$menu) {
     _manualdirectdebit_civix_insert_navigation_menu($menu, 'Administer/' . $directDebitMenuItem['name'], $menuItem);
   }
 }
+
+/**
+ * Implements hook_civicrm_postProcess().
+ *
+ */
+function manualdirectdebit_civicrm_postProcess($formName, &$form) {
+  $action = $form->getAction();
+
+  if ($formName == 'CRM_Contribute_Form_Contribution' && $action == CRM_Core_Action::ADD) {
+    $contributionId = (int) $form->getVar('_id');
+    $assignedContactId = (int) $form->getVar('_contactID');
+    $mandateId = getIdOfNewMandate($assignedContactId);
+    if ($form->_params['is_recur'] == 1) {
+      $recurrContributionId = $form->_params['contributionRecurID'];
+      saveDependencyBetweenContributionAndMandate($recurrContributionId, $mandateId);
+    }
+
+    saveMandateInContribution($contributionId, $mandateId);
+  }
+}
+
+/**
+ * Gets id of last inserted direct debit mandate
+ *
+ * @param $assignedContactId
+ *
+ * @return int
+ */
+function getIdOfNewMandate($assignedContactId) {
+  $tableName = 'civicrm_value_dd_mandate';
```

Comment:
The function name is getIdOfNewMandate, but it actually create a new mandate record which is confusing. The function should be renamed to createMandate() and createNewMandate implementation should be moved to it.